docker-compose-e2e-tests.yml

```diff
@@ -0,0 +1,67 @@
+version: "3.7"
+
+name: mit-open-e2e-tests
+
+services:
+  db:
+    image: postgres:12.17
+    ports:
+      - 5432:5432
+    environment:
+      - POSTGRES_PASSWORD=postgres
+    healthcheck:
+      test: ["CMD", "pg_isready", "-U", "postgres"]
+      interval: 5s
+      timeout: 5s
+      retries: 2
+
+  nginx:
+    image: nginx:1.25.3
+    ports:
+      - 8063:8063
+    links:
+      - web
+    volumes:
+      - ./config/nginx.conf:/etc/nginx/conf.d/web.conf
+    healthcheck:
+      test: ["CMD", "curl", "-f", "http://localhost:8063"]
+      interval: 30s
+      timeout: 10s
+      retries: 5
+
+  web:
+    build:
+      context: .
+    environment:
+      DATABASE_URL: postgres://postgres:postgres@db:5432/e2e_postgres # pragma: allowlist secret
+      PORT: 8061
+    env_file: .env.ci
+    depends_on:
+      db:
+        condition: service_healthy
+    command: > # pragma: allowlist secret
+      bash -c "export PGPASSWORD=postgres
+      && dropdb --if-exists -h db -U postgres e2e_postgres
+      && createdb -h db -U postgres e2e_postgres
+      && python3 manage.py collectstatic --noinput --clear
+      && python3 manage.py migrate --noinput
+      && python3 manage.py loaddata e2e_testing/fixtures/*.json
```

Comment:
This works great, but one potential worry for the future is that because data for all tests are loaded together, every test must be compatible with the data loaded for all other tests.  With python unit tests, any required data is typically created at the beginning of the test and is then auto discarded when it completes.  Not sure if there's a good way to do this for E2E testing, and that could be an enhancement for the future instead of for this PR, just figured it was worth mentioning.

Reply:
Yes, that's something to be aware of. There's some trade off for the convenience of using fixture files and a fresh database each run, though it's good to encourage writing tests that can also run in parallel without race conditions. If we setup and teardown data within each test it can be hard to spot conflicts when we parallelize tests to cut down the running time.